### PR TITLE
Builder skip Go binary validation if not used

### PR DESCRIPTION
**Pull request by foadnh**
**Description:** Builder skip Go binary validation if not used

As a follow-up on (#6009), when the builder is used with `-skip-compilation` and `-skip-get-modules` flags, it only generates distribution go codes. So there is no need for Go binary.
Isolated build environments like [Bazel](https://bazel.build/) or some CIs can skip passing Go binary to the builder.

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
I believe this change deserves a changelog entry.

**Comment by dmathieu** [comment]
Should a test be added?

**Comment by jpkrohling** [comment]
If you can add a test, it would be awesome.

**Comment by foadnh** [comment]
@jpkrohling changelog entry added.

**Comment by foadnh** [comment]
@dmathieu @jpkrohling I added a test for skipping Go binary validation. There are no tests for actually Go binary validation.

File: cmd/builder/internal/builder/config_test.go [hunk 1]
```diff
@@ -135,3 +135,14 @@ func TestNewDefaultConfig(t *testing.T) {
...
+
+func TestSkipGoValidation(t *testing.T) {
+	cfg := Config{
+		Distribution: Distribution{
+			Go: "invalid/go/binary/path",
+		},
+		SkipCompilation: true,
+		SkipGetModules:  true,
+	}
+	require.NoError(t, cfg.Validate())
```
Comment (dmathieu) [inline cmd/builder/internal/builder/config_test.go:147]: Maybe check that `c.Distribution.Go` is nil?

**Pull request merged**